lib/linguist/languages.yml

```diff
@@ -3289,6 +3289,21 @@ Jupyter Notebook:
   aliases:
   - IPython Notebook
   language_id: 185
+Justfile:
+  type: programming
+  aliases:
+  - Justfile
+  color: "#384d54"
+  tm_scope: source.justfile
+  extensions:
+  - ".justfile"
+  filenames:
+  - Justfile
+  - justfile
+  ace_mode: justfile
+  codemirror_mode: justfile
+  codemirror_mime_type: text/x-justfile
+  language_id: 20220919 #todo: how to set this?!
```

Comment:
[Step 4](https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-a-language) 😉

Reply:
I cannot run `bundle` on Ubuntu, even after installing the deps stated in the main readme and Contributing.md.
I also tried the docker from the main readme, cannot pull. Tried running [crazy-max/docker-linguist](https://github.com/crazy-max/docker-linguist) which does start but cannot run `bundle`.

Can you please help here?

Reply:
> I cannot run bundle on Ubuntu, even after installing the deps stated in the main readme and Contributing.md.

Please start a new [discussion](https://github.com/github/linguist/discussions) with details of what happens when you do and we can look into things independently of this PR.

I wouldn't use the docker images as they're for running Linguist, not developing it as they only include the bare minimums.

The error you have shown is because you're not running the container correctly. See https://github.com/crazy-max/docker-linguist#usage. Running it correctly wouldn't help with this either.

Reply:
If you really want to use Docker for developing Linguist, the [default container](https://github.com/microsoft/vscode-dev-containers/tree/main/containers/codespaces-linux) for Codespaces works a treat (I use it all the time) and it can also be used with VSCode remote containers.

Reply:
I've built the default container and can run it, producing the output as expected.
How do I run bundle on Ubuntu?

Reply:
😕 The command _is_ `bundle`. It would have been installed you ran `script/bootstrap`.

Reply:
Which default container?

The default container I referenced is _not_ a Linguist container. It's a generic dev container maintained by Microsoft and is used by Codespaces but can also be used with VSCode remote containers. 

The container created from the Dockerfile in this repo and the repo you referenced are, as I mentioned before, for _running_ Linguist from a published gem, _not_ developing Linguist from the source in this repo.

>  So does the default container have `bundle`?

From my previous comment (which I see is missing a word 🤦):

> It would have been installed [when] you ran `script/bootstrap`.

Reply:
Do you have a container to run bundle, it would greatly simplify things?

> It would have been installed [when] you ran script/bootstrap.
You didn't specify from where. The `bootstrap` seems broken in that regard.

Host:
```bash
foo@bar:~/git/linguist$ ./script/bootstrap 
./script/bootstrap: 14: bundle: not found
```

Container:
```bash
$ docker run -it --rm -v "$(pwd):/repo" --entrypoint sh crazymax/linguist:latest

/repo # ./script/bootstrap 
./script/bootstrap: line 14: bundle: not found
```

Reply:
> Do you have a container to run bundle, it would greatly simplify things?

No, but you can create one using the container I referenced before. Any full Ubuntu docker image would do the trick too. You can then manually install the dependencies as documented.

Regarding your host: it looks like you've not installed all the required dependencies, particularly the two Ruby packages (from the CONTRIBUTING.md file):

```
apt-get install cmake pkg-config libicu-dev docker.io ruby ruby-dev zlib1g-dev build-essential libssl-dev
```

A recent version of Ruby should include `bundle`. The bootstrap script will update it if it needs to.

As for the container: this is not expected to work. As I've said before, it is for running Linguist, not developing it.

Reply:
Nope.

<details>
  <summary>Commandline trace</summary>
```bash
foo@bar:~/git/linguist$ sudo apt update
Hit:1 http://dl.google.com/linux/chrome/deb stable InRelease
Hit:2 https://download.docker.com/linux/ubuntu focal InRelease
Get:3 http://security.ubuntu.com/ubuntu focal-security InRelease [114 kB]
Hit:4 http://ppa.launchpad.net/linuxuprising/shutter/ubuntu focal InRelease
Hit:5 http://packages.microsoft.com/repos/code stable InRelease
Hit:6 http://ppa.launchpad.net/unit193/encryption/ubuntu focal InRelease
Get:7 http://security.ubuntu.com/ubuntu xenial-security InRelease [99.8 kB]
Hit:8 http://ppa.launchpad.net/yubico/stable/ubuntu focal InRelease
Hit:9 http://de.archive.ubuntu.com/ubuntu focal InRelease
Get:10 http://de.archive.ubuntu.com/ubuntu focal-updates InRelease [114 kB]
Get:11 http://security.ubuntu.com/ubuntu focal-security/universe amd64 c-n-f Metadata [15.1 kB]
Get:12 http://de.archive.ubuntu.com/ubuntu focal-backports InRelease [108 kB]
Get:13 http://de.archive.ubuntu.com/ubuntu focal-backports/universe amd64 Packages [23.9 kB]
Get:14 http://de.archive.ubuntu.com/ubuntu focal-backports/universe i386 Packages [13.5 kB]
Fetched 488 kB in 2s (310 kB/s)
Reading package lists... Done
Building dependency tree
Reading state information... Done
66 packages can be upgraded. Run 'apt list --upgradable' to see them.

foo@bar:~/git/linguist$ sudo apt-get install cmake pkg-config libicu-dev docker.io ruby ruby-dev zlib1g-dev build-essential libssl-dev
Reading package lists... Done
Building dependency tree
Reading state information... Done
cmake is already the newest version (3.16.3-1ubuntu1).
pkg-config is already the newest version (0.29.1-0ubuntu4).
ruby is already the newest version (1:2.7+1).
ruby-dev is already the newest version (1:2.7+1).
build-essential is already the newest version (12.8ubuntu1.1).
libicu-dev is already the newest version (66.1-2ubuntu2.1).
libssl-dev is already the newest version (1.1.1f-1ubuntu2.16).
zlib1g-dev is already the newest version (1:1.2.11.dfsg-2ubuntu1.3).
docker.io is already the newest version (20.10.12-0ubuntu2~20.04.1).
0 upgraded, 0 newly installed, 0 to remove and 66 not upgraded.

foo@bar:~/git/linguist$ ./script/bootstrap
./script/bootstrap: 14: bundle: not found

foo@bar:~/git/linguist$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.4 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.4 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal


```
</details>

Reply:
If you've got the Ruby pkgs installed and still can't find `bundle` you can install it manually using `gem install bundler`

Reply:
🤦‍♂️ I've just noticed this is documented in the CONTRIBUTING.md file right under the Ubuntu requirements:

> The latest version of Bundler can be installed with `gem install bundler`.

I guess we both need to clean our glasses 😆

Reply:
> I guess we both need to clean our glasses laughing

You got me. :smile:

Reply:
On second thought, it's at the end of a new line as an afterthought.
It should get its own bash block in markdown, opened https://github.com/github/linguist/pull/6089 just now.